src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import duke.exception.DeadlineCommandUseException;
+import duke.exception.EventCommandUseException;
+import duke.exception.InvalidInputException;
+import duke.exception.ToDoCommandUseException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.Serializable;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+public class Parser implements Serializable {
+
+    public Parser() {
+        //empty constructor to initialize class objects
+    }
+
+    public void chat(String str, TaskList tasks) {
```

Comment:
Perhaps the chat method could be simplified, as to avoid it being a long method, which makes reading the code more difficult 😊

Reply:
yup, I see what you mean